src/main/java/net/neoforged/neoforge/common/damagesource/DamageContainer.java

```diff
@@ -0,0 +1,181 @@
+     *         any modifications are made.
+     */
+    public float getOriginalDamage() {
+        return originalDamage;
+    }
+
+    /**
+     * @return The damage source for this damage sequence
+     */
+    public DamageSource getSource() {
+        return source;
+    }
+
+    /**
+     * This sets the current damage value for the entity at the stage of the damage sequence in which it is set.
+     * Subsequent steps in the damage sequence will use and modify this value accordingly. If this is called in
+     * the final step of the sequence, this value will be applied against the entity's health.
+     *
+     * @param damage the amount to harm this entity at the end of the damage sequence
+     */
+    public void setNewDamage(float damage) {
+        this.newDamage = damage;
+    }
+
+    /**
+     * @return the current amount expected to be applied to the entity or used in subsequent damage calculations.
+     */
+    public float getNewDamage() {
+        return newDamage;
+    }
+
+    /**
+     * Adds a callback modifier to the vanilla damage reductions. Each function will be performed in sequence
+     * on the vanilla value at the time the {@link DamageContainer.Reduction} type is set by vanilla.
+     * <h4>Note: only the {@link LivingIncomingDamageEvent EntityPreDamageEvent}
+     * happens early enough in the sequence for this method to have any effect.</h4>
+     *
+     * @param type              The reduction type your function will apply to
+     * @param reductionFunction takes the current reduction from vanilla and any preceding functions and returns a new
+     *                          value for the reduction. These are always executed in insertion order. if sequence
+     *                          matters, use {@link net.neoforged.bus.api.EventPriority} to order your function.
+     */
+
+    public void addModifier(Reduction type, IReductionFunction reductionFunction) {
+        this.reductionFunctions.computeIfAbsent(type, a -> new ArrayList<>()).add(reductionFunction);
+    }
+
+    /**
+     * @return The damage blocked during the {@link LivingShieldBlockEvent}
+     */
+    public float getBlockedDamage() {
+        return blockedDamage;
+    }
+
+    /**
+     * @return The durability applied to the applicable shield after {@link LivingShieldBlockEvent}
+     *         returned a successful block
+     */
+    public float getShieldDamage() {
+        return shieldDamage;
+    }
+
+    /**
+     * Explicitly sets the invulnerability ticks after the damage has been applied.
+     *
+     * @param ticks Ticks of invulnerability after this damage sequence
+     */
+    public void setPostAttackInvulnerabilityTicks(int ticks) {
+        this.invulnerabilityTicksAfterAttack = ticks;
+    }
+
+    /**
+     * @return The number of ticks this entity will be invulnerable after damage is applied.
+     */
+    public int getPostAttackInvulnerabilityTicks() {
+        return invulnerabilityTicksAfterAttack;
+    }
```

Comment:
I'd make these @return tags inline {@return}.
Remember not to end them with a dot and to start them with lowercase.